src/rgw/driver/rados/rgw_datalog.h

```diff
@@ -466,6 +466,10 @@ class RGWDataChangesLog {
 			     std::string>>
   list_entries(const DoutPrefixProvider* dpp, int shard, int max_entries,
 	       std::string marker);
+  int list_entries_checked(const DoutPrefixProvider *dpp, int shard, int max_entries,
+			   std::vector<rgw_data_change_log_entry>& entries,
+			   std::string_view marker, std::string* out_marker,
+			   bool* truncated, optional_yield y);
```

Comment:
i don't see why we want an unchecked version of this function

the admin api `RGWOp_DATALog_List` used by data sync is still calling the unchecked `datalog_rados->list_entries()`

Reply:
Agreed. Sorry about the delay.